R/nth.R

```diff
@@ -0,0 +1,38 @@
+nth_default <- function(x, default = NA) {
+  vec_assert(default, size = 1)
+  vec_cast(default, x)
+}
+
+#' @export
+nth <- function(x, n, default = NA) {
+  x_size <- vec_size(x)
+
+  if (x_size == 0) {
+    return(nth_default(x, default))
+  }
```

Comment:
I think this is nicely captured by `n == 0 || n > x_size || n < -x_size` and might not be required

Reply:
Thanks